addons/product_configurator/static/src/js/product_configurator_field.js

```diff
@@ -0,0 +1,152 @@
+                        value: data.custom_value,
+                    }
+                }
+            ),
+            edit: edit,
+            options: this.props.options,
+            save: async (mainProduct, optionalProducts) => {
+                await this.applyProduct(this.props.record, mainProduct);
+            },
+            discard: () => {},
+            ...this.additionalParams(),
+        };
+    },
+
+    additionalParams() {
+        if(this.props.options?.hideQty){
+            return {companyId: this.props.record.model.root.data.company_id[0]};
+        }
+        const product_uom_id = this.props.options?.productUomIdFieldName ? this.props.options.productUomIdFieldName : 'product_uom';
+        return {
+            quantity: this.props.record.data.product_uom_qty,
+            productUOMId: this.props.record.data[product_uom_id][0],
+            companyId: this.props.record.model.root.data.company_id[0]
+        };
+    },
+
+    onEditConfiguration() {
+        if (this.isConfigurableLine) {
+            this._editLineConfiguration();
+        } else {
+            this._editProductConfiguration();
+        }
+    },
+
+    get isConfigurableTemplate() {
+        return super.isConfigurableTemplate || this.props.record.data.is_configurable_product;
+    },
+
+    _editProductConfiguration() {
+        super._editProductConfiguration(...arguments);
+        if (this.props.record.data.is_configurable_product) {
+            this._openProductConfigurator(true);
+        }
+    },
+
+    async applyProduct(record, product) {
+        // handle custom values & no variants
+        const contextRecords = [];
+        for (const ptal of product.attribute_lines) {
+            const selectedCustomPTAV = ptal.attribute_values.find(
+                ptav => ptav.is_custom && ptav.id === ptal.selected_attribute_value_id
+            );
+            if (selectedCustomPTAV) {
+                contextRecords.push({
+                    default_custom_product_template_attribute_value_id: selectedCustomPTAV.id,
+                    default_custom_value: ptal.customValue,
+                });
+            };
+        }
+
+        const proms = [];
+        proms.push(record.data.product_custom_attribute_value_ids.createAndReplace(contextRecords));
+
+        const noVariantPTAVIds = product.attribute_lines.filter(
+            ptal => ptal.create_variant === "no_variant" && ptal.attribute_values.length > 1
+        ).map(ptal => ptal.selected_attribute_value_id);
+
+        await Promise.all(proms);
+        const default_qty_key = this.props.options?.qtyName ? this.props.options.qtyName : 'product_uom_qty';
+        await record.update({
+            product_id: [product.id, product.display_name],
+            [default_qty_key]: product.quantity,
+            product_no_variant_attribute_value_ids: [x2ManyCommands.set(noVariantPTAVIds)],
+        });
+    }
+});
```

Comment:
This is a util that was custom-created to suit sales needs. You don't even handle custom attribute values in MRP. This method can exist here as a hook and should be defined entirely in each sub-module, as each model is different.

Reply:
We actually do want to handle the custom attributes in MRP. That is the main point of that PR (we had to manufacture the custom variant without being able to create/modify them)😉